### PR TITLE
[#164631266] Set Keychain accessible attribute

### PR DESCRIPTION
**Pull request by fpersico**

### CODE REVIEW

File: ts/utils/keychain.ts [hunk 1]
```diff
@@ -17,7 +17,9 @@ const PIN_KEY = "PIN";
  * Saves the provided PIN in the Keychain
  */
 export async function setPin(pin: PinString): Promise<boolean> {
-  return await Keychain.setGenericPassword(PIN_KEY, pin);
+  return await Keychain.setGenericPassword(PIN_KEY, pin, {
```
Comment (cloudify) [inline ts/utils/keychain.ts]: Can you create a wrapper of `setGenericPassword` here in `utils` that adds the `accessible: Keychain.SecAccessible.WHEN_UNLOCKED_THIS_DEVICE_ONLY` option, and use that in both places? Also link to the relevant documentation describing `WHEN_UNLOCKED_THIS_DEVICE_ONLY`

**Comment by cloudify** [comment]
@fpersico `[WIP]` tag is deprecated, use Draft Pull Requests as described [in the handbook](https://github.com/teamdigitale/io-handbook/blob/master/development-workflow.md#pull-requests)

**Comment by fpersico** [comment]
@cloudify There is still no way to move the PR to draft after the creation. 
I remembered having to do an additional test so i have added [WIP].

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Remember to fix the PR title by removing WIP wording when ready
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Please include a description of your PR changes.
  </td>
    </tr>
  </tbody>
</table>



## Affected stories

  * 🌟 [#164631266](https://www.pivotaltracker.com/story/show/164631266): Keychain items stored with weak access control attribute

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: ts/utils/keychain.ts [hunk 2]
```diff
@@ -13,11 +13,28 @@ import { PinString } from "../types/PinString";
...
+ * https://developer.apple.com/documentation/security/ksecattraccessibleafterfirstunlock
+ */
+export async function setGenericPasswordWithDefaultAccessibleOption(
+  username: string,
+  password: string,
+  options?: Options
+) {
+  return Keychain.setGenericPassword(username, password, {
+    ...options,
+    accessible: Keychain.SecAccessible.WHEN_UNLOCKED_THIS_DEVICE_ONLY
```
Comment (cloudify) [inline ts/utils/keychain.ts]:
```suggestion
    // The data in the keychain item can be accessed only while the device is unlocked by the user.
    // This is recommended for items that need to be accessible only while the application is in the foreground. Items 
    // with this attribute do not migrate to a new device. Thus, after restoring from a backup of a different device, 
    // these items will not be present.
    accessible: Keychain.SecAccessible.WHEN_UNLOCKED_THIS_DEVICE_ONLY
```

**Pull request merged**